### PR TITLE
IGNITE-19587: Sql. Remove execution-related part from IgniteTable.

### PR DESCRIPTION
**Pull request by lowka**
Removes a reference to InternalTable from IgniteTableImpl.
Extracts implementation of `UpdateableTable` out of IgniteTableImpl.
Moves IgniteTable::toRow into a separate interface.

Introduces `ExecutionDepedencyResolver` that traverses an operator tree and requests components required by execution engine for each table it encounters via `ExecutableTableRegistry `. Implementation of `ExecutableTableRegistry`  internally calls all the necessary components to retrieve `InternalTable`, `UpdateableTable` in asynchronous matter.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/thread/StripedThreadPoolExecutor.java [hunk 1]
```diff
@@ -105,7 +106,7 @@ public void execute(Runnable cmd) {
      * @throws NullPointerException       if the task is {@code null}.
      */
     public CompletableFuture<?> submit(Runnable task, int idx) {
-        return CompletableFuture.runAsync(task, execs[threadId(idx)]);
+        return CompletableFuture.runAsync(task, commandExecutor(idx));
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/internal/thread/StripedThreadPoolExecutor.java:109]: `@param idx` is missed in javadoc.
Would you please fix this?

Comment (lowka) [reply modules/core/src/main/java/org/apache/ignite/internal/thread/StripedThreadPoolExecutor.java:110]: Done.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java [hunk 2]
```diff
@@ -114,6 +116,9 @@ public class SqlQueryProcessor implements QueryProcessor {
     /** Size of the cache for query plans. */
     private static final int PLAN_CACHE_SIZE = 1024;
 
+    /** Size of the table access cache. */
+    public static final int TABLE_CACHE_SIZE = 1024;
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java:120]: why `public`?

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutableTable.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Execution related APIs of a table.
+ */
+public interface ExecutableTable {
+
+    /**
+     * Returns read API.
+     */
+    InternalTable table();
```
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutableTable.java:30]: will do.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/QueryTaskExecutor.java [hunk 4]
```diff
@@ -45,4 +45,13 @@ public interface QueryTaskExecutor extends LifecycleAware, Executor {
...
     CompletableFuture<?> submit(UUID qryId, long fragmentId, Runnable qryTask);
+
+    /**
+     * Returns an executor that is responsible for the given query fragment.
+     *
+     * @param qryId      Query ID.
+     * @param fragmentId Fragment ID.
+     * @return  An executor.
+     */
+    Executor fragmentExecutor(UUID qryId, long fragmentId);
```
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/QueryTaskExecutor.java:56]: Done.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutableTable.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutableTable.java]: Done.

**Pull request merged**